assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -162,6 +162,12 @@ const SurveyQuestionMultiSelect = ( {
 							>
 								<Checkbox
 									checked={ answer.selected }
+									disabled={
+										hasMaximumNumberOfChoices &&
+										! answer.selected
+											? true
+											: undefined
+									}
```

Comment:
Per my last comment, this condition could be a bit simpler as a boolean rather than a ternary to use `undefined` (no attribute) which seems like an odd coupling to the test.